### DISCUSSION TITLE
Works with more than Omnik

### DISCUSSION THREAD

**Discussion by ghost** [robbinjanssen/home-assistant-omnik-inverter#94] [General]
Hi, just wanted to say a big thank you for this lib. I've got a Hosola 1500TL inverter and it works perfectly. The inverter uses status.js. Not sure if Hosola and Omnik are the same company, but they at least seem to use the same firmware.

Having solar panel data in the energy dashboard is some next level stuff, thanks again.

**Comment by robbinjanssen** [comment]
I will add it to the readme :) thanks for sharing!

**Reply by dn-gt** [reply]
Hello, it works also with the Huayu HY600 Pro converter using the HTML connection and I was able to add it to Home Assistant without issues (JSON and Javascript approach did not work). What I'm getting is the AC power in Watt, cumulated energy for the day and total cumulated in kWh.
The inverter also has a data logger with much more detailed data, e.g. DC power, current and voltage on module level. Do you see any possibility, to get this data?

**Reply by robbinjanssen** [reply]
Great to hear! I'll add it to the list.

Currently we only pass through whatever the JS/JSON/HTML output provides us. @MarijnS95 is working on a TCP implementation which provides the data you are talking about, however we don't know if this will work for all inverters :) See https://github.com/klaasnicolaas/python-omnikinverter/pull/134 for the progress.
